### DISCUSSION TITLE
How to control the colour of a light

### DISCUSSION THREAD

**Discussion by heisenberg2980** [mikejac/node-red-contrib-google-smarthome#468] [Q&A]
I have just found this amazing project and after playing around for a while I have been able to set everything up and turn some lights on/off. Now I am trying to control the brightness and the colour of the lights, and while the brightness was pretty simple by using the value payload.brightness to update the brightness_pct of the light, the colour seems more complicated as the value I see in spectrumRgb is an integer instead of the 3 RBG values exposed by the light, is there an easy way to convert these values in both directions so I can ask google to change the colour of the light and also update google with the current light colour every time it changes?

**Comment by heisenberg2980** [comment]
I have been able to control the colour by using the HSV instead of the RGB

**Comment by Caprico85** [comment]
In v0.4.9, the Light example was extended to show how to handle RGB lights.

![image](https://github.com/mikejac/node-red-contrib-google-smarthome/assets/2081806/67adf1b2-a43c-4964-9b97-8d5849fb7314)
